tests/unit/experimental/globus_app/test_globus_app.py

```diff
@@ -211,6 +218,29 @@ def test_user_app_get_authorizer():
     assert authorizer.access_token == "mock_access_token"
 
 
+def test_user_app_get_authorizer_clears_cache_when_adding_scope_requirements():
+    client_id = "mock_client_id"
+    memory_storage = MemoryTokenStorage()
+    memory_storage.store_token_data_by_resource_server(_mock_token_data_by_rs())
+    config = GlobusAppConfig(token_storage=memory_storage)
+    user_app = UserApp("test-app", client_id=client_id, config=config)
+
+    initial_authorizer = user_app.get_authorizer("auth.globus.org")
+    assert isinstance(initial_authorizer, AccessTokenAuthorizer)
+    assert initial_authorizer.access_token == "mock_access_token"
+
+    # We should've cached the authorizer from the first call
+    assert user_app.get_authorizer("auth.globus.org") == initial_authorizer
```

Comment:
When checking for cached instances, it seems like this should be an identity check.

```suggestion
    assert user_app.get_authorizer("auth.globus.org") is initial_authorizer
```

Reply:
Sure, good callout.